tools/analysis_tools/analyze_logs.py

```diff
@@ -50,55 +52,22 @@ def plot_curve(log_dicts, args):
         epochs = list(log_dict.keys())
         for j, metric in enumerate(metrics):
             print(f'plot curve of {args.json_logs[i]}, metric is {metric}')
-            if metric not in log_dict[epochs[args.interval - 1]]:
+            if metric not in log_dict[epochs[int(args.eval_interval) - 1]]:
                 raise KeyError(
-                    f'{args.json_logs[i]} does not contain metric {metric}')
-
-            if args.mode == 'eval':
```

Comment:
Why the content of mode `eval` has been deleted?

Reply:
Original implementation will raise a `TypeError`.

Reply:
The `json` file does not content `mode` key. If users want to plot evaluation metric, we can set `args.eval` to True.

Reply:
when i run python tools/analysis_tools/analyze_logs.py plot_curve 79.7\ 496\ 432/20230902_001556/vis_data/20230902_001556.json --keys loss --eval
i encounter :ValueError: x and y must have same first dimension, but have shapes (399,) and (400,)
whats this error?